src/Xamarin.Android.Build.Tasks/Utilities/MarshalMethodsClassifier.cs

```diff
@@ -0,0 +1,388 @@
+using System;
+using System.Collections.Generic;
+
+using Java.Interop.Tools.Cecil;
+using Java.Interop.Tools.JavaCallableWrappers;
+using Java.Interop.Tools.TypeNameMappings;
+using Microsoft.Android.Build.Tasks;
+using Microsoft.Build.Utilities;
+using Mono.Cecil;
+
+namespace Xamarin.Android.Tasks
+{
+#if ENABLE_MARSHAL_METHODS
+	public sealed class MarshalMethodEntry
+	{
+		public TypeDefinition TopType             { get; }
+		public MethodDefinition NativeCallback    { get; }
+		public MethodDefinition Connector         { get; }
+		public MethodDefinition RegisteredMethod  { get; }
+		public MethodDefinition ImplementedMethod { get; }
+		public FieldDefinition CallbackField      { get; }
+		public string JniTypeName                 { get; }
+		public string JniMethodName               { get; }
+		public string JniMethodSignature          { get; }
+
+		public MarshalMethodEntry (TypeDefinition topType, MethodDefinition nativeCallback, MethodDefinition connector, MethodDefinition
+		                           registeredMethod, MethodDefinition implementedMethod, FieldDefinition callbackField, string jniTypeName,
```

Comment:
Indentation convention: an "extra" tab-stop vs. the body of the method, with tabs.  Please do not align with `(`, and start each line with a type:

```csharp
public MarshalMethodEntry (TypeDefinition topType, MethodDefinition nativeCallback, MethodDefinition connector,
[tab][tab]MethodDefinition registeredMethod, MethodDefinition implementedMethod, FieldDefinition callbackField,
[tab][tab]string jniTypeName, string jniName, string jniSignature)
{
[tab]// contents…
}
```

Rationale: if (when?) this method is renamed, all subsequent lines won't need to be altered to re-align with `(`.